tw-tasks-core/src/main/java/com/transferwise/tasks/helpers/CoreMetricsTemplate.java

```diff
@@ -179,11 +181,21 @@ public void registerFailedNextEventTimeChange(String taskType, ZonedDateTime fro
   }
 
   @Override
-  public void registerTaskGrabbingResponse(String bucketId, String taskType, int priority, ProcessTaskResponse processTaskResponse) {
-    meterCache.counter(METRIC_TASKS_TASK_GRABBING, TagsSet.of(TAG_TASK_TYPE, taskType,
-            TAG_BUCKET_ID, bucketId, TAG_PRIORITY, String.valueOf(priority), TAG_GRABBING_RESPONSE, processTaskResponse.getResult().name(),
-            TAG_GRABBING_CODE, processTaskResponse.getCode() == null ? "UNKNOWN" : processTaskResponse.getCode().name()))
-        .increment();
+  public void registerTaskGrabbingResponse(String bucketId, String taskType, int priority, ProcessTaskResponse processTaskResponse,
+      Instant taskTriggeredAt) {
+
+    TagsSet tags = TagsSet.of(
+        TAG_TASK_TYPE, taskType,
+        TAG_BUCKET_ID, bucketId,
+        TAG_PRIORITY, String.valueOf(priority),
+        TAG_GRABBING_RESPONSE, processTaskResponse.getResult().name(),
+        TAG_GRABBING_CODE, processTaskResponse.getCode() == null ? "UNKNOWN" : processTaskResponse.getCode().name()
+    );
+
+    meterCache.counter(METRIC_TASKS_TASK_GRABBING, tags).increment();
+
+    long millisSinceTaskTriggered = System.currentTimeMillis() - taskTriggeredAt.toEpochMilli();
```

Comment:
I just think if you add a sane default here, even affecting quality of metric that way, it would be safer.

We don't know, when someone changes that logic and null pops here by accident.